core/shared/utils/bh_log.c

```diff
@@ -79,3 +82,63 @@ bh_print_time(const char *prompt)
 
     last_time_ms = curr_time_ms;
 }
+
+#ifdef linux
```

Comment:
This ifdef makes me think whether it actually should be in `shared` folder? If it's shared with all `posix`  or `linux` platforms, maybe we should have `posix` or `linux` shared folder instead?

Reply:
😆  

OK. `#ifdef` represents that the implementation with */proc/self/status* only workable on Linux-like system. And still looking for solution on other platforms. Eventually, all platforms should have its own implementation to collect memory consumption information. In my mind, `bh_print_vm()` should be a common feature.

Reply:
Is it better to use `#if defined(BH_PLATFORM_LINUX) || defined(BH_PLATFORM_DARWIN)`?

Reply:
Agree to add the platform related code in platform related folder, for example, define API like `os_print_virt_mem` in `core/shared/platform/include/platform_api_extension.h`, and implement it in `core/shared/platform/common/posix/posix_malloc.c` for posix like platforms currently, and implement it for Windows in the future.